src/utils.ts

```diff
@@ -133,33 +133,330 @@ export namespace Utils {
+   * to a signer (accountIDs) for verification to succeed. If verification succeeds,
+   * a list of signers that were found is returned, not including the server account ID.
+   *
+   * Signers that are not prefixed as an address/account ID strkey (G...) will be ignored.
+   *
+   * Errors will be raised if:
+   *  - The transaction is invalid according to ReadChallengeTx.
+   *  - No client signatures are found on the transaction.
+   *  - One or more signatures in the transaction are not identifiable as the
+   *    server account or one of the signers provided in the arguments.
+   *
+   * @see [SEP0010: Stellar Web Authentication](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0010.md).
+   * @function
+   * @memberof Utils
+   * @param {string} challengeTx SEP0010 challenge transaction in base64.
+   * @param {string} serverAccountID The server's stellar account (public key).
+   * @param {string} networkPassphrase The network passphrase, e.g.: 'Test SDF Network ; September 2015'.
+   * @param {string[]} accountIDs The signers public keys. This list should contain the public keys for all signers that have signed the transaction.
+   * @returns {string[]} The list of signers public keys that have signed the transaction, excluding the server account ID.
+   * @example
+   *
+   * import { Networks, Transaction, Utils }  from 'stellar-sdk';
+   *
+   * const serverKP = Keypair.random();
+   * const clientKP1 = Keypair.random();
+   * const clientKP2 = Keypair.random();
+   *
+   * // Challenge, possibly built in the server side
+   * const challenge = Utils.buildChallengeTx(
+   *   serverKP,
+   *   clientKP1.publicKey(),
+   *   "SDF",
+   *   300,
+   *   Networks.TESTNET
+   * );
+   *
+   * // clock.tick(200);  // Simulates a 200 ms delay when communicating from server to client
+   *
+   * // Trsansaction gathered from a challenge, possibly from the client side
+   * const transaction = new Transaction(challenge, Networks.TESTNET);
+   * transaction.sign(clientKP1, clientKP2);
+   * const signedChallenge = transaction
+   *         .toEnvelope()
+   *         .toXDR("base64")
+   *         .toString();
+   *
+   * // The result below should be equal to [clientKP1.publicKey(), clientKP2.publicKey()]
+   * Utils.verifyChallengeTxSigners(signedChallenge, serverKP.publicKey(), Networks.TESTNET, threshold, clientKP1.publicKey(), clientKP2.publicKey());
+   */
+  export function verifyChallengeTxSigners(
+    challengeTx: string,
+    serverAccountID: string,
+    networkPassphrase: string,
+    ...accountIDs: string[]
+  ): string[] {
+    // Read the transaction which validates its structure.
+    const { tx } = readChallengeTx(
+      challengeTx,
+      serverAccountID,
+      networkPassphrase,
+    );
+
+    // Ensure the server account ID is an address and not a seed.
+    const serverKP = Keypair.fromPublicKey(serverAccountID); // can throw 'Invalid Stellar public key'
```

Comment:
I couldn't really make the code break there, so I haven't added any new test cases for it, but I'm now re-throwing with an updated error message just to make things clearer in case it ever happens.